ansible_collections/arista/avd/plugins/action/eos_validate_state_runner.py

```diff
@@ -63,6 +63,22 @@ def run(self, tmp=None, task_vars=None):
         if not isinstance(skipped_tests, list):
             raise AnsibleActionFail(f"'skipped_tests' must be a list of dictionnaries, got {skipped_tests}.")
 
+        # Check device filter
+        device_filter = self._task.args.get("device_filter")
+        if not isinstance(device_filter, (str, list)):
+            raise AnsibleActionFail(f"'device_filter' must be a string or a list of string, got {device_filter}.")
+
+        device_filter = [device_filter] if isinstance(device_filter, str) else device_filter
+
+        inventory_manager = task_vars["hostvars"]._inventory
+        host = inventory_manager.get_host(hostname)
+        host_groups = {group.get_name() for group in host.get_groups()}
+
+        if not set(device_filter).intersection(host_groups):
```

Comment:
LGTM
Just If we can add the support to filter by some specific devices or regex of devices name.